src/unix/syscall.c

```diff
@@ -834,6 +857,17 @@ sysreturn open_internal(filesystem fs, tuple cwd, const char *name, int flags,
         }
     }
 
+    if (do_missing_files) {
```

Comment:
I gather that this is a feature that would only be switched on in the rare case that some libraries / support files can't be found, and thus does not need to be efficient in any way, correct? If it were ever enabled by default, this implementation might (?) cause a significant performance penalty, given that this is adding an entry for every file - whether found or not - and then removing it from the missing_files vector for found files. Also, this implementation requires a linear search (vs, for example, a table-based implemenation).

Is it possible to make this add files only on the condition that a file is not found? Or is it written this way because a file might be not found and then later created?

Reply:
You are correct, it's only intended to be a debugging feature. I had to implement like this because glibc tries several different paths for each dylib (e.g. /lib/x86_64-linux-gnu, /lib/x86_64-linux-gnu/x86_64, /usr/lib/x86_64-linux-gnu, etc) so if a file isn't in the first path then it's added to the list and if it's found later in a different directory then it's not really missing. It would definitely be more efficient to use a table. I originally thought it would be useful to preserve the order in which files are attempted to be open, but now I've tried it a few times I don't know that order is really that important.

Reply:
Yes, this makes sense and it looks good as it is. I don't think there's any need to change it to make a debug-only feature slightly more efficient.